### PR TITLE
Bugfix for detect-engine.luajit-states

### PR DESCRIPTION
**Pull request by togge**
detect-engine is a list, and luajit-states was looked up as a map.

### CODE REVIEW

**Comment by inliniac** [comment]
@jasonish can you check this?

File: src/detect-lua.c [hunk 1]
```diff
@@ -190,12 +190,18 @@ int DetectLuajitSetupStatesPool(int num, int reloads)
...
     if (luajit_states == NULL) {
-        int cnt = 0;
-        char *conf_val = NULL;
+        intmax_t cnt = 0;
+        ConfNode *denode = NULL;
+        ConfNode *decnf = ConfGetNode("detect-engine");
+        if (decnf != NULL) {
+            TAILQ_FOREACH(denode, &decnf->head, next) {
+                if (strcmp(denode->val, "luajit-states") == 0) {
+                    ConfGetChildValueInt(denode, "luajit-states", &cnt);
```
Comment (jasonish) [inline src/detect-lua.c]: Maybe "break" after this.

**Comment by jasonish** [comment]
Looks fine. This is the best way to deal with the YAML in its current form.

**Comment by inliniac** [comment]
Merged through #1699, thanks!

**Pull request closed without merging**